pkg/filter21/ast/tree.go

```diff
@@ -30,6 +30,9 @@ func (id *Identifier) Equal(ident Identifier) bool {
 
 // String returns the string representation for the Identifier
 func (id *Identifier) String() string {
+	if id.Field == nil {
```

Comment:
Worth also doing this if `id == nil`?

Reply:
Go is the weirdest language, isn't it? I forgot about this behavior. https://go.dev/play/p/cMCUho0IyIl

Why, oh why, are you allowed to call methods on a typed nil?

Anyway, good callout -- I've added a guard.